tests/testthat/test-pkgdown.R

```diff
@@ -0,0 +1,29 @@
+test_that("can parse pkgdown lists", {
+  expect_equal(
+    pkgdepends_parse(character()),
+    list(repos = NULL, refs = character()))
+  expect_equal(
+    pkgdepends_parse("foo"),
+    list(repos = NULL, refs = "foo"))
+  expect_equal(
+    pkgdepends_parse(c("foo", "bar")),
+    list(repos = NULL, refs = c("foo", "bar")))
+  expect_equal(
+    pkgdepends_parse(c("# a comment", "foo", "bar")),
+    list(repos = NULL, refs = c("foo", "bar")))
+  expect_equal(
+    pkgdepends_parse(c("foo", "", "bar")),
+    list(repos = NULL, refs = c("foo", "bar")))
+  expect_equal(
+    pkgdepends_parse(c(" foo", "   ", "bar ")),
+    list(repos = NULL, refs = c("foo", "bar")))
+  expect_equal(
+    pkgdepends_parse(c("repo::https://mrc-ide.r-universe.dev", "foo")),
+    list(repos = "https://mrc-ide.r-universe.dev", refs = "foo"))
+  expect_error(
+    pkgdepends_parse("foo-bar"),
+    "Failed to parse some package references")
+  expect_error(
+    pkgdepends_parse(c("foo", "bar", "foo-bar")),
+    "Failed to parse some package references")
+})
```

Comment:
What should the behaviour be if we have multiple repos? 
(Currently, we end up with repos=c("repo1", "repo2") and refs = c("ref1", "ref2")

Are we able to express that one ref comes from one repo, and another from another? (Eg, in the case where we have different versions of packages in different repos, and we might pick and mix - which is all very terrible, but malariasimulation and friends seem to have that issue.)

Reply:
One other thing - I guess a particular branch of a package is ref1@branch1 - and is treated in exactly the same way (hence no need for a song+dance about it)

Reply:
And (finally!) - is there any value in checking for duplicate refs (with/without `@branch`) ?

Reply:
as discussed, this is all bits that pkgdepends will try and do the right thing on, we'll try not to get too involved with its logic